diff -up libsolv-0.7.29/src/poolarch.c.1~ libsolv-0.7.29/src/poolarch.c
--- libsolv-0.7.29/src/poolarch.c.1~	2024-04-30 15:29:06.000000000 +0200
+++ libsolv-0.7.29/src/poolarch.c	2024-06-14 01:00:09.993458385 +0200
@@ -23,6 +23,7 @@
 static const char *archpolicies[] = {
 #if defined(FEDORA) || defined(MAGEIA)
   "x86_64_v4",	"x86_64_v4:x86_64_v3:x86_64_v2:x86_64:athlon:i686:i586:i486:i386",
+  "znver1",	"znver1:x86_64_v3:x86_64_v2:x86_64:athlon:i686:i586:i486:i386",
   "x86_64_v3",	"x86_64_v3:x86_64_v2:x86_64:athlon:i686:i586:i486:i386",
   "x86_64_v2",	"x86_64_v2:x86_64:athlon:i686:i586:i486:i386",
   "x86_64",	"x86_64:athlon:i686:i586:i486:i386",
@@ -32,6 +33,7 @@ static const char *archpolicies[] = {
   "x86_64_v2",	"x86_64_v2:x86_64:i686:i586:i486:i386",
   "x86_64",	"x86_64:i686:i586:i486:i386",
 #endif
+  "znver1_32",	"znver1_32:athlon:i686:i586:i486:i386",
   "i686",	"i686:i586:i486:i386",
   "i586",	"i586:i486:i386",
   "i486",	"i486:i386",
