diff -uraN libsolv-0.7.35/CMakeLists.txt omv-libsolv-0.7.35/CMakeLists.txt
--- libsolv-0.7.35/CMakeLists.txt	2025-08-04 11:14:31.000000000 +0200
+++ omv-libsolv-0.7.35/CMakeLists.txt	2025-10-26 20:21:03.251188430 +0100
@@ -288,7 +288,7 @@
 	CMAKE_POP_CHECK_STATE ()
 ENDMACRO (check_linker_flag)
 check_linker_flag("-Wl,--as-needed" HAVE_LINKER_AS_NEEDED)
-check_linker_flag("-Wl,--version-script=${CMAKE_SOURCE_DIR}/src/libsolv.ver" HAVE_LINKER_VERSION_SCRIPT)
+check_linker_flag("-Wl,--undefined-version -Wl,--version-script=${CMAKE_SOURCE_DIR}/src/libsolv.ver" HAVE_LINKER_VERSION_SCRIPT)
 
 # should create config.h with #cmakedefine instead...
 FOREACH (VAR HAVE_STRCHRNUL HAVE_FOPENCOOKIE HAVE_FUNOPEN WORDS_BIGENDIAN
diff -uraN libsolv-0.7.35/ext/CMakeLists.txt omv-libsolv-0.7.35/ext/CMakeLists.txt
--- libsolv-0.7.35/ext/CMakeLists.txt	2025-08-04 11:14:31.000000000 +0200
+++ omv-libsolv-0.7.35/ext/CMakeLists.txt	2025-10-26 20:21:50.853325921 +0100
@@ -160,7 +160,7 @@
 SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
 ENDIF (NOT MSVC)
 IF (HAVE_LINKER_VERSION_SCRIPT)
-SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_FLAGS} -Wl,--version-script=${CMAKE_SOURCE_DIR}/ext/libsolvext.ver")
+SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_FLAGS} -Wl,--undefined-version -Wl,--version-script=${CMAKE_SOURCE_DIR}/ext/libsolvext.ver")
 ENDIF (HAVE_LINKER_VERSION_SCRIPT)
 
 IF (DISABLE_SHARED)
diff -uraN libsolv-0.7.35/src/CMakeLists.txt omv-libsolv-0.7.35/src/CMakeLists.txt
--- libsolv-0.7.35/src/CMakeLists.txt	2025-08-04 11:14:31.000000000 +0200
+++ omv-libsolv-0.7.35/src/CMakeLists.txt	2025-10-26 20:22:28.823300618 +0100
@@ -49,7 +49,7 @@
 ENDIF (WIN32)
 
 IF (HAVE_LINKER_VERSION_SCRIPT)
-SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_FLAGS} -Wl,--version-script=${CMAKE_SOURCE_DIR}/src/libsolv.ver")
+SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${LINK_FLAGS} -Wl,--undefined-version -Wl,--version-script=${CMAKE_SOURCE_DIR}/src/libsolv.ver")
 ENDIF (HAVE_LINKER_VERSION_SCRIPT)
 
 IF (DISABLE_SHARED)
